src/main/java/com/zendesk/maxwell/SynchronousBootstrapper.java

```diff
@@ -0,0 +1,200 @@
+		try ( Connection connection = context.getConnectionPool().getConnection() ) {
+			// This update resets all rows of incomplete bootstraps to their original state.
+			// These updates are treated as fresh bootstrap requests and trigger a restart
+			// of the bootstrap process from the beginning.
+			String sql = "update maxwell.bootstrap set started_at = NULL where is_complete = 0 and started_at is not NULL";
+			connection.prepareStatement(sql).execute();
+		}
+	}
+
+	private Table findTable(String tableName, Database database) {
+		Table table = database.findTable(tableName);
+		if ( table == null )
+			throw new RuntimeException("Couldn't find table " + tableName);
+		return table;
+	}
+
+	private Database findDatabase(Schema schema, String databaseName) {
+		Database database = schema.findDatabase(databaseName);
+		if ( database == null )
+			throw new RuntimeException("Couldn't find database " + databaseName);
+		return database;
+	}
+
+	private void ensureTable(String tableName, Database database) {
+		findTable(tableName, database);
+	}
+
+	private ResultSet getAllRows(String databaseName, String tableName, Connection connection) throws SQLException, InterruptedException {
+		Statement statement = createBatchStatement(connection);
+		return statement.executeQuery(String.format("select * from %s.%s", databaseName, tableName));
+	}
+
+	private Statement createBatchStatement(Connection connection) throws SQLException, InterruptedException {
+		Statement statement = connection.createStatement();
+		statement.setFetchSize(context.getConfig().bootstrapperBatchFetchSize);
+		return statement;
+	}
+
+	private void setBootstrapRowToStarted(RowMap startBootstrapRow, Connection connection) throws SQLException {
+		String sql = "update maxwell.bootstrap set started_at=NOW() where id=?";
+		PreparedStatement preparedStatement = connection.prepareStatement(sql);
+		preparedStatement.setLong(1, ( Long ) startBootstrapRow.getData("id"));
+		preparedStatement.execute();
+	}
+
+	private void setBootstrapRowToCompleted(int insertedRows, RowMap startBootstrapRow, Connection connection) throws SQLException {
+		String sql = "update maxwell.bootstrap set is_complete=1, inserted_rows=?, completed_at=NOW() where id=?";
+		PreparedStatement preparedStatement = connection.prepareStatement(sql);
+		preparedStatement.setInt(1, insertedRows);
+		preparedStatement.setLong(2, ( Long ) startBootstrapRow.getData("id"));
+		preparedStatement.execute();
+	}
+
+	private void setRowValues(RowMap row, ResultSet resultSet, Table table) throws SQLException, IOException {
+		Iterator<ColumnDef> columnDefinitions = table.getColumnList().iterator();
+		int columnIndex = 1;
+		while ( columnDefinitions.hasNext() ) {
+			ColumnDef columnDefinition = columnDefinitions.next();
+			row.putData(columnDefinition.getName(), columnDefinition.getObjectFromResultSet(resultSet, columnIndex));
+			++columnIndex;
+		}
```

Comment:
Agree about the sanity check. In order for this to work, the users of mysql would have to make not to do any DDL statements while the bootstrap is in progress.